tests/pipeline/test_manifest.py

```diff
@@ -0,0 +1,341 @@
+        mock_get_version = mocker.patch.object(Helm, "get_version")
+        mock_get_version.return_value = Version(major=3, minor=12, patch=0)
+        return mock_get_version
+
+    @pytest.fixture(autouse=True)
+    def helm(self, mock_get_version: MagicMock) -> Helm:
+        return Helm(helm_config=HelmConfig())
+
+    def test_default_config(self, mock_execute: MagicMock):
+        result = runner.invoke(
+            app,
+            [
+                "manifest",
+                str(RESOURCE_PATH / "custom-config/pipeline.yaml"),
+                "--defaults",
+                str(RESOURCE_PATH / "no-topics-defaults"),
+                "--environment",
+                "development",
+            ],
+            catch_exceptions=False,
+        )
+        mock_execute.assert_called_with(
+            [
+                "helm",
+                "template",
+                "resources-custom-config-app2",
+                "bakdata-streams-bootstrap/streams-app",
+                "--namespace",
+                "development-namespace",
+                "--values",
+                ANY,
+                "--version",
+                "2.9.0",
+                "--timeout",
+                "5m0s",
+                "--wait",
+            ],
+        )
+        assert result.exit_code == 0, result.stdout
+
+    def test_custom_config(self, mock_execute: MagicMock):
+        result = runner.invoke(
+            app,
+            [
+                "manifest",
+                str(RESOURCE_PATH / "custom-config/pipeline.yaml"),
+                "--defaults",
+                str(RESOURCE_PATH / "no-topics-defaults"),
+                "--config",
+                str(RESOURCE_PATH / "custom-config"),
+                "--environment",
+                "development",
+            ],
+            catch_exceptions=False,
+        )
+        mock_execute.assert_called_with(
+            [
+                "helm",
+                "template",
+                "resources-custom-config-app2",
+                "bakdata-streams-bootstrap/streams-app",
+                "--namespace",
+                "development-namespace",
+                "--values",
+                ANY,
+                "--version",
+                "2.9.0",
+                "--timeout",
+                "5m0s",
+                "--wait",
+                "--api-versions",
+                "2.1.1",
+            ],
+        )
+        assert result.exit_code == 0, result.stdout
+
+    def test_python_api(self):
+        steps = kpops.manifest(
+            RESOURCE_PATH / "custom-config/pipeline.yaml",
+            defaults=RESOURCE_PATH / "no-topics-defaults",
+            output=False,
+            environment="development",
+        )
+        assert len(steps) == 2
+        manifests = steps[0]
+        assert len(manifests) == 1
+        assert isinstance(manifests[0], KubernetesManifest)
+        assert manifests[0] == {
```

Comment:
How about we read the manifest content from a YAML file? IMO, it makes the test file smaller and easier to read.

Reply:
good idea! Probably we could even use a snapshot to benefit from automatic re-generation when we make changes